lib/ansible/modules/cloud/openstack/os_nova_host_aggregate.py

```diff
@@ -167,6 +167,9 @@ def main():
             if aggregate is None:
                 changed = False
             else:
+                if hosts:
```

Comment:
Does it require to provide the list of hosts explicitly? The example does not provide the hosts input for the deleting the aggregate.  I would expect to ansible to remove all the nodes from the aggregate by itself, instead of user input.

Reply:
Yes, the list of the hosts should be provided.
When you are using variables for the hosts, the hosts could be added or deleted depending by the state provided.

This is the default behavior of openstack.
It can't remove the hosts from the aggregate without getting the hosts provided.

I don't think ansible module should override this behavior.
It supports the openstack api capabilities and should not change the logic of openstack.

Reply:
> It supports the openstack api capabilities and should not change the logic of openstack.

IMO, ansible enables to use the openstack APIs in a simplified manner, rather than providing a one-to-one mapping of all openstack cli commands. I may be wrong, i would love to hear from experts. If it is approved with the current approach, it would be good to update example with 'hosts' mentioned for 'absent' scenario too.